docs/source/components.rst

```diff
@@ -0,0 +1,332 @@
+==========
+Components
+==========
+
+This document describes all components or "steps" the new Django serializers
+will have. They are closely related to existing concepts of ``django.forms`` and
+``django.core.serializers`` and borrows ideas from Django Rest Framework (DRF).
+
+Each section explains its scope and what exactly it should and shouldn't do and
+know about. It also tries to include a note about the specific step is
+currently realized in Django or DRF and gives hints how that step could be
+implemented in the future.
+
+The components are supposed to be independent from each other and it should be
+easy to customize one of it without having to touch or even braking any other
+component. Ideally you should also be able to replace a component entirely, e.g.
+with an external library.
+
+Additionally introspection of the components should be possible even within
+components. This ensures that components can be independent from each other
+while still being generic enough to make reuse of code easier.
+
+
+Construction
+============
+
+The construction step determines which fields will be used in the following
```

Comment:
`determines the shape and logic of the resulting serializer`? The fact we have "fields" is actually just one part of the whole thing - this helps "cross-field" logic become a first level concept.